### ISSUE TITLE
Development efficiency documentation needed

### ISSUE THREAD

**Issue by melissjs** [dashpay/dash-website#386]
Great documentation for setting up, the only thing that is missing is information on more efficient development. It takes my machine about 8 minutes to start the site and after that incremental watch changes take about 3+ minutes. In config.yml, I commented out every language but "en" after the initial startup and then re-rendering changes only takes about 7 seconds. I realize this is hacky and there is probably a more elegant solution but this is what worked for me. Another issue I ran into was not being about to commit changes to my repo due to htmlproof errors. I had to temporarily remove htmlproof from pre-commit in package.json as a work around.

**Comment by chuckwilliams37** [comment]
Good catch, Melissa - and the language trick is something that typically gets communicated to everyone who works on the site very early as "tribal knowledge".  The htmlproof intentionally prevents commits because otherwise the site won't pass through our Travis CI Build checks.

It's been used so-far as a checkpoint to help developers conform to the team's basic needs. 

It's been a thorn in the web-team's side how long it takes to build. A big part of it is image generation/resizing & as you've discovered languages. For the language portion - we intend to change the process so that blog posts don't necessarily sit on the dev boxes and get processed along w/ builds. For the image resizing - I think it's configured so that if you run it locally & develop - the images only get updated at startup, or during command-line build invocations. I'm open to other ideas!

**Comment by melissjs** [comment]
Thanks so much for responding Chuck. I am new to a lot of these technologies so it took me a little time to figure out but it wasn't so bad. Maybe a quick note on the language tribal knowledge in the documentation would help. I think something else might be going on with my htmlproof errors bc it is generating lots (and lots!) of errors for things I didn't create. Maybe somehow my build is missing something it is looking for...

**Comment by chuckwilliams37** [comment]
I haven't looked in a while - but my suspicion is that it something simple like mistreated ampersands in blog-posts or some such simple thing that are causing issues. That was something I had to deal with quite a bit back when I was still working on the Web site full time. I wouldn't be overly concerned by it, it's probably not your fault.
